src/webhook/validation/oneagent_test.go

```diff
@@ -290,3 +292,162 @@ func TestImageFieldSetWithoutCSIFlag(t *testing.T) {
 	})
 
 }
+
+func TestOneAgentVolumeStorageReadOnlyModeConflict(t *testing.T) {
+	testcases := []struct {
+		testName     string
+		creator      func(featureFlag string, featureFlagValue bool, oaEnvValue bool) *dynatracev1beta1.DynaKube
+		featureFlag  string
+		featureValue bool
+	}{
+		{"conflicting settings in cloudNative",
+			createCloudNativeFullstackDynaKube,
+			dynatracev1beta1.AnnotationFeatureReadOnlyOneAgent, true},
+		{"conflicting settings in hostMonitoring",
+			createHostMonitoringDynaKube, dynatracev1beta1.AnnotationFeatureReadOnlyOneAgent, true},
+		{"conflicting settings in cloudNative with deprecated feature flag",
+			createCloudNativeFullstackDynaKube, dynatracev1beta1.AnnotationFeatureDisableReadOnlyOneAgent, false},
+		{"conflicting settings in hostMonitoring with deprecated feature flag",
+			createHostMonitoringDynaKube, dynatracev1beta1.AnnotationFeatureDisableReadOnlyOneAgent, false},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.testName, func(t *testing.T) {
+			assertDeniedResponse(t,
+				[]string{errorVolumeStorageReadOnlyModeConflict},
+				tc.creator(tc.featureFlag, tc.featureValue, true),
+				&defaultCSIDaemonSet)
+		})
+	}
+}
```

Comment:
```suggestion

func TestOneAgentVolumeStorageReadOnlyModeConflict(t *testing.T) {
	t.Run("enabling read-only mode with cloud-native is denied", func(t *testing.T) {
		assertDeniedResponse(t,
			[]string{errorVolumeStorageReadOnlyModeConflict},
			createCloudNativeFullstackDynaKube(dynatracev1beta1.AnnotationFeatureReadOnlyOneAgent, true, true),
			&defaultCSIDaemonSet)
	})
	
	t.Run("enabling read-only mode with deprecated flag and cloud-native is denied", func(t *testing.T) {
		assertDeniedResponse(t,
			[]string{errorVolumeStorageReadOnlyModeConflict},
			createCloudNativeFullstackDynaKube(dynatracev1beta1.AnnotationFeatureReadOnlyOneAgent, false, true),
			&defaultCSIDaemonSet)
	})
	
	t.Run("enabling read-only mode with host-monitoring is denied", func(t *testing.T) {
		assertDeniedResponse(t,
			[]string{errorVolumeStorageReadOnlyModeConflict},
			createHostMonitoringDynaKube(dynatracev1beta1.AnnotationFeatureReadOnlyOneAgent, true, true),
			&defaultCSIDaemonSet)
	})
	
	t.Run("enabling read-only mode with deprecated flag and host-monitoring is denied", func(t *testing.T) {
		assertDeniedResponse(t,
			[]string{errorVolumeStorageReadOnlyModeConflict},
			createHostMonitoringDynaKube(dynatracev1beta1.AnnotationFeatureDisableReadOnlyOneAgent, false, true),
			&defaultCSIDaemonSet)
	})
}
```

Reply:
How does that improve the code? Its just needless code duplication.

Reply:
its more readable and there is barely any code duplication.
looping over test functions makes the code needlessly complex

Reply:
No it's not. The test cases are collected in a single table all at one place, where you can easily spot the the difference with one glimpse instead of having to compare 4 almost similar code snippets. The test execution logic is there once instead of 4 times, which means it's more maintainable. And the "complexity" of a simple loop is tolerable.

Reply:
To understand a test-case, in your case one has to remember the method itself, e.g. `createHostMonitoringDynaKube`, the struct and what is inside `tc.creator`, then think about how `tc.creator` is called. After these steps it can be understood how the code works.

In the other case, how execution is performed is immediately obvious to a programmer.

Reply:
If that does not convince you, then think about it this way: The way we write tests is standardized in the contribution guidelines. These guidelines may not always align with personal preferences, like in this case: is readability better then less code duplication? I would argue that in this case the answer is subjective. However, the guidelines exist so that the code base has a unified style. That would be another reason to implement the tests in the suggested way, because it aligns better with the majority of the other unit tests

Reply:
You're right, I'm not convinced ;-)

First of all, throwing the guidelines into the ring seems like a killer argument to me, so I'm going to ignore that for now. 

Second, to understand the test cases (what's tested) you merely have to have a look at the test table. How the test is performed, isn't actually that important at first. The table summarizes the covered test cases in a kind of abstracted way. Only if you want to go a bit deeper you can have a look at the execution logic. So you can approach the test code in a two-step manner - abstract to detail.
What aids the readability is, that the whole test method fits on a screen and it probably still would with double the number of tests. But I may rethink the naming of the "creator" member, though. It's probably to generic.

Generally speaking, table tests tend to need a bit of getting-used to in the first place (was my experience too), but they can really clean up test code and avoid lot's of tedious maintenance work when the tested code changes. Had this quite often in the past, that test code grew and grew (which mostly is a good thing as it means more tests) and once you made a tiny change to a used struct, a signature etc. you had to adopt tests in dozens of locations. So if guidelines don't allow them, it's maybe time to challenge the guidelines.

Reply:
i see your points. In that case I'd like to see a well defined struct, maybe not anonymous but local to the test function with good member names. Maybe after this PR is done I'd suggest you create a PR to amend the guidelines so table tests are also standardized

Reply:
Baptized structs and added test cases.